.github/workflows/clippy.yaml

```diff
@@ -0,0 +1,55 @@
+name: Release Static Checks
+
+on:
+  pull_request:
+    branches:
+      - msft-main
+
+jobs:
+  clippy:
+    name: Run Clippy on Rust Components
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@v4
+
+      - name: Install Dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install -y libdevmapper-dev clang llvm
+
+      - name: Install Rust Toolchain
+        uses: dtolnay/rust-toolchain@stable
+        with:
+          components: clippy
```

Comment:
We should install Rust the way that upstream does it (install_rust.sh).

Regardless, we might not want to pick up Rust's latest version here, as it could break us. We should probably fetch the version from versions.yaml (and ensure the version that's there is the one we want).